src/Groups/cosets.jl

```diff
@@ -65,13 +74,72 @@ function show(io::IO, x::GroupCoset)
    a = GAP.gap_to_julia(GAP.Globals.StringView(x.H.X))
    b = GAP.gap_to_julia(GAP.Globals.StringView(x.repr.X))
    if x.side == :right
-      print(io, a, " * ", b)
+      print(io, "Right coset   ", a, " * ", b)
    else
-      print(io, b, " * ", a)
+      print(io, "Left coset   ", b, " * ", a)
    end
    return nothing
 end
 
+"""
+    isleft(c::GroupCoset)
+Return whether the coset `c` is a left coset of its acting domain.
+"""
+isleft(c::GroupCoset) = c.side == :left
+
+"""
+    isright(c::GroupCoset)
+Return whether the coset `c` is a right coset of its acting domain.
+"""
+isright(c::GroupCoset) = c.side == :right
+
+"""
+    as_right_coset(c::GroupCoset)
+Return a coset (either left or right) as a right coset. 
+"""
+function as_right_coset(c::GroupCoset)
+   if c.side == :right return c
+   else return right_coset(c.H^(c.repr^-1), c.repr)
+   end
+end
+
+"""
+    as_left_coset(c::GroupCoset)
+Return a coset (either left or right) as a left coset. 
+"""
+function as_left_coset(c::GroupCoset)
+   if c.side == :left return c
+   else return left_coset(c.H^(c.repr), c.repr)
+   end
+end
+
+Base.:*(H::GAPGroup, g::GAPGroupElem) = right_coset(H,g)
+Base.:*(g::GAPGroupElem, H::GAPGroup) = left_coset(H,g)
+
+function Base.:*(c::GroupCoset, y::GAPGroupElem)
+   @assert y in c.G "element not in the group"
+   if c.side == :right
+      return right_coset(c.H, c.repr*y)
+   else
+      return left_coset(c.H^y, c.repr*y)
+   end
+end
+
+function Base.:*(y::GAPGroupElem, c::GroupCoset)
+   @assert y in c.G "element not in the group"
+   if c.side == :left
+      return left_coset(c.H, y*c.repr)
+   else
+      return right_coset(c.H^(y^-1), y*c.repr)
+   end
+end
+
+function Base.:*(c::GroupCoset, d::GroupCoset)
+   c1=as_right_coset(c)
+   d1=as_left_coset(d)
+   return double_coset(c1.H, c1.repr*d1.repr, d1.H)
```

Comment:
I would suggest to only support this when c is a left coset and d is a right coset.

Reply:
Done.